app/(dashboard)/enroll/[repositoryId]/page.tsx

```diff
@@ -1,90 +1,39 @@
+"use client";
+
 import { DashboardHeader } from "@/components/header";
 import { DashboardShell } from "@/components/shell";
 import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
-import Image from "next/image";
-import Link from "next/link";
+import { TRepository, ZRepository } from "@/types/repository";
+import { useEffect, useState } from "react";
 
+import { getRepositoryDataAction } from "./actions";
 import EnrollmentStatusBar from "./enrollmentStatusBar";
-import FB1 from "./fb-view-1.webp";
-import FB2 from "./fb-view-2.webp";
 
-const frombricksFeatures = [
-  {
-    icon: "📲",
-    description: "Create conversion-optimized surveys with our no-code editor with several question types.",
-  },
-  {
-    icon: "📚",
-    description: "Choose from a variety of best-practice templates.",
-  },
-  {
-    icon: "👩🏻",
-    description:
-      "Launch and target your surveys to specific user groups without changing your application code.",
-  },
-  {
-    icon: "🔗",
-    description: "Create shareable link surveys.",
-  },
-  {
-    icon: "👨‍👩‍👦",
-    description: "Invite your team members to collaborate on your surveys.",
-  },
-  {
-    icon: "🔌",
-    description: "Integrate Formbricks with Slack, Notion, Zapier, n8n, and more.",
-  },
-  {
-    icon: "🔒",
-    description: "All open source, transparent, and self-hostable.",
-  },
-];
+export default function RepositoryDetailPage({ params }) {
+  const [repository, setRepository] = useState<TRepository | null>(null);
```

Comment:
All the logic for making the repository call including the server action isn't necessary. When it's a server component you can just use
```
const repository = getRepositoryById(params.repositoryId)
```
and you can get rid of of the server action, the states, the useEffect and the loading states. If you need a loading state, use the nextjs `loading.ts` ( https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming )